lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+--[[
+
+    dtutils.lua - common darktable lua functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+dtutils = {}
+
+local dt = require "darktable"
+
+local log = require "lib/libLog"
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("dtutils",dt.configuration.config_dir.."/lua/")
+
+local function _(msgid)
+    return gettext.dgettext("dtutils", msgid)
+end
+
+--[[
+  NAME
+    dutils.split_filepath - split a filepath into parts
+
+  SYNOPSIS
+    result = dtutils.split_filepath(path_string)
+      filepath - path and filename
+
+  DESCRIPTION
+    split_filepath splits a filepath into the path, filename, basename and filetype and puts
+    that in a table
+
+  RETURN VALUE
+    result - a table containing the path, filename, basename, and filetype
+
+  ERRORS
+
+
+
+]]
```

Comment:
script_manager already has a routine to pull block comments and reformat them.  I can probably pull that and wrap some code around it to create a tool to pull the library docs